proto/tendermint/state/types.proto

```diff
@@ -11,13 +11,28 @@ import "tendermint/types/params.proto";
 import "tendermint/version/types.proto";
 import "google/protobuf/timestamp.proto";
 
-// ABCIResponses retains the responses
-// of the various ABCI calls during block processing.
-// It is persisted to disk for each height before calling Commit.
-message ABCIResponses {
-  repeated tendermint.abci.ResponseDeliverTx deliver_txs = 1;
-  tendermint.abci.ResponseEndBlock           end_block   = 2;
-  tendermint.abci.ResponseBeginBlock         begin_block = 3;
+// LegacyABCIResponses retains the responses
+// of the legacy ABCI calls during block processing.
+// Note ReponseDeliverTx is renamed to ExecTxResult but they are semantically the same
+// Kept for backwards compatibility for versions prior to v0.38
+message LegacyABCIResponses {
+  repeated tendermint.abci.ExecTxResult deliver_txs = 1;
+  ResponseEndBlock                      end_block   = 2;
+  ResponseBeginBlock                    begin_block = 3;
+}
+
+// ResponseBeginBlock is kept for backwards compatibility for versions prior to v0.38
+message ResponseBeginBlock {
+  repeated tendermint.abci.Event events = 1
+      [(gogoproto.nullable) = false, (gogoproto.jsontag) = "events,omitempty"];
+}
+
+// ResponseEndBlock is kept for backwards compatibility for versions prior to v0.38
+message ResponseEndBlock {
+  repeated tendermint.abci.ValidatorUpdate validator_updates       = 1 [(gogoproto.nullable) = false];
+  tendermint.types.ConsensusParams         consensus_param_updates = 2;
+  repeated tendermint.abci.Event events                            = 3
+      [(gogoproto.nullable) = false, (gogoproto.jsontag) = "events,omitempty"];
```

Comment:
Is there a reason why we need to keep these? We decided to removed them in `v0.36.x`